operations/operations.py

```diff
@@ -0,0 +1,185 @@
+                                copy=False)
+    masked_median_v = np.ma.median(masked_values, axis=0)
+    if neighbors.variances is not None:
+        masked_median_var = np.ma.median(neighbors.variances, axis=0)
+        np.ma.array(neighbors.variances, mask=edges_mask.values, copy=False)
+        return sc.Variable(dims=neighbors.dims[1:],
+                           values=masked_median_v,
+                           variances=masked_median_var)
+    return sc.Variable(dims=neighbors.dims[1:], values=masked_median_v)
+
+
+def median_from_adj_pixels(data):
+    """
+    Applies a median across 8 neighboring pixels (plus centre value)
+    for data with 'x' and 'y' dimensions (at least).
+    Result will calculate median from slices across additional dimensions.
+
+    For example if there is a tof dimension in addition to x, and y,
+    for each set of neighbours the returned median will take the median tof value in the neighbour group.
+    """
+    fill = np.finfo(data.values.dtype).min
+    has_variances = not data.variances is None
+    container = sc.Variable(['neighbor'] + data.dims,
+                            shape=[
+                                9,
+                            ] + data.shape,
+                            variances=has_variances)
+    container['neighbor', 0] = data
+    container['neighbor', 1] = _shift(data, "x", True, fill)
+    container['neighbor', 2] = _shift(data, "x", False, fill)
+    container['neighbor', 3] = _shift(data, "y", True, fill)
+    container['neighbor', 4] = _shift(data, "y", False, fill)
+    container['neighbor', 5:7] = _shift(container['neighbor', 1:3], "y", True,
+                                        fill)
+    container['neighbor', 7:9] = _shift(container['neighbor', 1:3], "y", False,
+                                        fill)
+
+    edges_mask = sc.less_equal(container, sc.Variable(value=fill,
+                                                      variance=fill))
+    return _median(container, edges_mask, dim='neighbor')
+
+
+def _calc_adj_spectra(center_spec_num, bank_width, num_spectra):
+    col_positions = [-1, 0, 1]  # Take relative positions of col
+
+    # We calculate along a row at a time, rather than a col at a time
+    top_row_center = (center_spec_num - bank_width)
+    bottom_row_center = (center_spec_num + bank_width)
+
+    neighbour_spec = []
+
+    # Top / Middle / Bottom row
+    neighbour_spec.extend(i + top_row_center for i in col_positions)
+    neighbour_spec.extend(i + center_spec_num for i in col_positions)
+    neighbour_spec.extend(i + bottom_row_center for i in col_positions)
+
+    def is_valid(spec_num):
+        # Do not include OOB and the centre spectra
+        return 0 <= spec_num < num_spectra and spec_num != center_spec_num
+
+    neighbour_spec = [i for i in neighbour_spec if is_valid(i)]
+    return neighbour_spec
+
+
+def generate_neighbouring_spectra(num_spectra: int, bank_width: int):
```

Comment:
This function and the one above do not seem required any more since you refactored the neighbor median code?